### PR TITLE
feat(protocol-designer): form fields restyle 

### LINKED ISSUE #1284: Condense form fields in advanced settings
As a user I would like to not be visually overwhelmed with form fields I don't intend to use. 

--

### Part 1
- [x] When checkboxes with attached for fields are unchecked, don't display those form fields. 
- [x] Turn change tip into a dropdown
- [x] Update 2nd column's fields to match the illustration (no well order, new placeholders for flow rate & tip position)
- [x] Optimize styles for smallest supported screen width

### Part 2 (TODO - break out into new separate ticket!)

- For now the advanced settings are in two columns: (1) Technique (2) Change tip, well order, flow rate.
- [ ] The two columns are evenly spaced initially, but then the 2nd moves right to accomodate changing width in column 1. 

## Design
image

### PR DESCRIPTION
**Pull request by IanLondon**
## overview

Closes #1284 (I am going to split "Part 2" into its own ticket) and #1322

Addresses #1294 in that it removes the serialize name field, but that ticket isn't done with b/c the correct name isn't used for serialization, it still defaults to "Sample 1" etc. So 1294 will be a separate quick PR.

New styles for both Step Edit Form (all step types) & Ingredient Edit Form (both add new ingred, and edit existing ingred by clicking Ingred in sidebar)

## changelog

* pd: cleanup & componentize StepEditForm
* pd: cleanup Form.css
* complib: style FormGroup

## review requests

https://s3-us-west-2.amazonaws.com/opentrons-protocol-designer/pd_condense-form-fields-1284-part-1/index.html

* style review! 
    * Step Edit Form (all step types)
    * Ingredient Edit Form (both add new ingred, and edit existing ingred by clicking Ingred in sidebar)
* code review. Sorry about doing 2 issues in one PR. IngredientEditForm was reliant upon the same CSS as StepEditForm so it was as much work to temporarily fix the pre-design-update IngredientEditForm as it was to fix it and use the updated design in #1322.

### CODE REVIEW

File: protocol-designer/src/components/IngredientPropertiesForm.js [hunk 1]
```diff
@@ -30,19 +30,22 @@ type FieldProps = {|
   label?: string,
   units?: string,
   error?: string,
-  placeholder?: string
+  placeholder?: string,
+  className?: string
```
Comment (mcous) [inline protocol-designer/src/components/IngredientPropertiesForm.js]: Nothing to do with this PR, but @Kadee80 and I were discussing turning on all trailing commas in eslint to improve these sorts of diffs. Any thoughts on that?

File: protocol-designer/src/components/IngredientPropertiesForm.js [hunk 2]
```diff
@@ -203,64 +203,73 @@ class IngredientPropertiesForm extends React.Component<Props, State> {
...
+              <Field accessor='description' />
+            </FormGroup>
           </div>
         </form>
 
         <div className={styles.button_row}>
-          <FlatButton /* disabled={TODO: validate input here} */
+          <PrimaryButton onClick={onCancel}>Cancel</PrimaryButton>
+
+          <PrimaryButton
```
Comment (mcous) [inline protocol-designer/src/components/IngredientPropertiesForm.js]: Should this button:

1. Be inside the `<form>`?
2. Have `type="submit"`?

File: protocol-designer/src/components/StepEditForm/formFields.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+type CheckboxRowProps = {
+  label?: string,
+  formConnector: FormConnector<*>,
+  checkboxAccessor: string,
+  children?: ?React.Node,
+  className?: string
+}
+export function CheckboxRow (props: CheckboxRowProps) {
+  const {formConnector, checkboxAccessor, label, className} = props
```
Comment (mcous) [inline protocol-designer/src/components/StepEditForm/formFields.js]: Not really the point of this PR, but I see it a lot so I do want to bring it up:

I'm not sure I understand the benefit of passing in `formConnector` and making the component retrieve `onChange` and `value` itself vs. just passing in `value` and `onChange` as props. On a related note, the `name` attribute of `<input>` seems more canonical than `accessor`

File: protocol-designer/src/components/StepEditForm/index.js [hunk 4]
```diff
@@ -42,79 +51,101 @@ export default function StepEditForm (props: Props) {
...
-              </FormGroup>
-              <div className={styles.field_row}>
-                <CheckboxField label='Delay' {...formConnector('dispense--delay--checkbox')} />
-                <InputField units='m' {...formConnector('dispense--delay-minutes')} />
-                <InputField units='s' {...formConnector('dispense--delay-seconds')} />
-              </div>
-              <div className={styles.field_row}>
-                <CheckboxField label='Blow out' {...formConnector('dispense--blowout--checkbox')} />
-                <DropdownField className={styles.column_2_3}
+      <div className={formStyles.form}>
```
Comment (mcous) [inline protocol-designer/src/components/StepEditForm/index.js]: Should this be a `<form>`?

Applies to all of these if cases

File: protocol-designer/src/components/StepEditForm/index.js [hunk 5]
```diff
@@ -42,79 +51,101 @@ export default function StepEditForm (props: Props) {
...
+  )
+
   const buttonRow = <div className={styles.button_row}>
-    <FlatButton onClick={props.onClickMoreOptions}>MORE OPTIONS</FlatButton>
-    <PrimaryButton onClick={props.onCancel}>CANCEL</PrimaryButton>
+    <FlatButton className={styles.more_options_button} onClick={props.onClickMoreOptions}>
+      MORE OPTIONS
+    </FlatButton>
+    <PrimaryButton className={styles.cancel_button} onClick={props.onCancel}>CANCEL</PrimaryButton>
     <PrimaryButton disabled={!props.canSave} onClick={props.onSave}>SAVE</PrimaryButton>
```
Comment (mcous) [inline protocol-designer/src/components/StepEditForm/index.js]: If the node below is made a `<form>`, this should be `type=submit` I think

File: protocol-designer/src/components/IngredientPropertiesForm.js (hunk 1 above)
Comment (IanLondon) [reply protocol-designer/src/components/IngredientPropertiesForm.js]: trailing commas always required? seems good

**Pull request merged**